unlock-app/src/pages/_app.js

```diff
@@ -5,80 +5,32 @@ import { WalletService, Web3Service } from '@unlock-protocol/unlock-js'
 import TagManager from 'react-gtm-module'
 import configure from '../config'
 
-import GlobalStyle from '../theme/globalStyle'
 import { ConfigContext } from '../utils/withConfig'
 import { WedlockServiceContext } from '../contexts/WedlocksContext'
 import WedlockService from '../services/wedlockService'
 import ProviderContext from '../contexts/ProviderContext'
-import Authenticate from '../components/interface/Authenticate'
+import GlobalWrapper from '../components/interface/GlobalWrapper'
 import '../index.css'
 
 const config = configure()
 const wedlockService = new WedlockService(config.services.wedlocks.host)
-
 class UnlockApp extends App {
   constructor(props, context) {
     super(props, context)
-    this.state = {
-      provider: null,
-    }
 
     if (!config.isServer) {
       if (config.env === 'prod' && config.tagManagerArgs) {
         TagManager.initialize(config.tagManagerArgs)
       }
```

Comment:
This can be turned into a functional component. Even though it is extending, it's not using any specific class-based component APIs.

https://nextjs.org/docs/advanced-features/custom-app
Import the app prop type from next/app